spec/services/notifications/new_badge_achievement/send_spec.rb

```diff
@@ -0,0 +1,50 @@
+require "rails_helper"
+
+RSpec.describe Notifications::NewBadgeAchievement::Send, type: :service do
+  let(:badge_achievement) { create(:badge_achievement) }
+
+  def expected_json_data
+    {
+      user: Notifications.user_data(badge_achievement.user),
+      badge_achievement: {
+        badge_id: badge_achievement.badge_id,
+        rewarding_context_message: badge_achievement.rewarding_context_message,
+        badge: {
+          title: badge_achievement.badge.title,
+          description: badge_achievement.badge.description,
+          badge_image_url: badge_achievement.badge.badge_image_url
+        }
+      }
+    }.to_json
+  end
+
+  it "creates a notification" do
+    expect do
+      described_class.call(badge_achievement)
+    end.to change(Notification, :count).by(1)
+  end
+
+  it "creates a notification for the badge achievement user" do
+    described_class.call(badge_achievement)
+    expect(Notification.last.user).to eq(badge_achievement.user)
```

Comment:
Are you sure relying on the last record is 100% safe? In this case, it's possible to just test the notification that is returned by the `call` method, and it's more explicit in my opinion.

Reply:
You're right, `.last` can result in a condition where the notification retrieved is not exactly the notification you were hoping to test. 

I pushed a change